stdlib/public/Python/Python.swift

```diff
@@ -39,12 +39,17 @@ typealias OwnedPyObjectPointer = PyObjectPointer
 final class PyReference {
   private var pointer: OwnedPyObjectPointer
 
-  init(owning pointer: OwnedPyObjectPointer) {
+  // This `PyReference`, once deleted, will make no delta change to the
+  // python object's reference count. It will however, retain the reference for
+  // the lifespan of this object.
+  init(retaining pointer: OwnedPyObjectPointer) {
```

Comment:
I think that the previous two names had the flipped meaning and should definitely be changed. However, I don’t think “retaining:” is the accurate word here because it’s not merely just retaining the object—it is creating a new reference and will release it later. Something along the lines of “creating a new reference” would be better. Maybe get rid of the argument label?

Reply:
I'm not convinced. Also, that would mean PythonObject would have a 1 argument init() on this PyReference. I think we want some form of label. "retaining:" is fairly accurate because the constructed reference will retain the pointer for the duration of the class's lifetime but make no other modifications to the reference count.

Reply:
I don’t think I agree. PyReference’s initializer in Swift implies the creation of a new heap object, and this new object is responsible for keeping itself alive and transitively preserves the lifetime of what self depends on (the init argument in this case). It is a value-preserving conversion. I consider the preservation of the Python object pointer’s lifetime part of PyReference’s semantics. 

Value-preserving conversions do not need an argument label, according to Swift API Design Guidelines. 

> In initializers that perform value preserving type conversions, omit the first argument label. 

The reason we had a label in the borrowing initializer is that the conversion would create an unbalanced reference (similar to ‘Unmanaged’), in which case, a label is necessary for clarity and predictability.

Also similar to Unmanaged, which is the only API in stdlib that mentions “retain” and “release”, calling this value-preserving initializer “retaining” implies unbalanced reference counting to me as a Swift programmer, but it is not unbalanced. 

One might argue that this is not a value-preserving conversion, which could make a sense since this is a C-pointer. But even in that case, “retaining:” is still implying an unbalanced retain to me here.

I think I’m fine with either no argument label, or an argument label that describes this common “adding a new reference” operation without saying “retain”, something like “newReference:”.

Reply:
I agree that `retaining` sounds like explicit unbalanced retaining counting, which isn't the actual behavior.

I support the idea of using a label-less initializer instead!